xamarin/realm-tutorial-dotnet/TaskPage.xaml.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using RealmTemplateApp.Models;
+using Realms;
+using Xamarin.Forms;
+using System.ComponentModel;
+
+namespace RealmTemplateApp
+{
+    public partial class TaskPage : ContentPage
+    {
+        private Realm taskRealm;
+        private ObservableCollection<Task> _tasks = new ObservableCollection<Task>();
+
+        public ObservableCollection<Task> MyTasks
+        {
+            get
+            {
+                return _tasks;
+            }
+        }
+
+        public TaskPage()
+        {
+            InitializeComponent();
+        }
+
+        protected override async void OnAppearing()
+        {
+            WaitingLayout.IsVisible = true;
+            try
+            {
+                taskRealm = await Realm.GetInstanceAsync();
+                SetUpTaskList();
+            }
+            catch (Exception ex)
+            {
+                await DisplayAlert("Error Fetching Tasks", ex.Message, "OK");
+            }
+            base.OnAppearing();
+        }
+
+        private void SetUpTaskList()
+        {
+            if (_tasks != null || _tasks.Count == 0)
+            {
+                WaitingLayout.IsVisible = true;
+                _tasks = new ObservableCollection<Task>(taskRealm.All<Task>().ToList());
```

Comment:
I got sucked down this path as it is/was considered a best practice for binding in a view, but I totally see how this is an anti-pattern for Realm. Fixed.

Reply:
Yes, this is actually the point. It's not necessarily very clear for developers how to use Realm effectively, and that it can simplify things quite a lot!